pkg/trace/agent/run.go

```diff
@@ -39,7 +39,9 @@ func Run(ctx context.Context) {
 	}
 
 	cfg, err := config.Load(flags.ConfigPath)
-	if err != nil {
+	// config.ErrMissingAPIKey is handled later in the method
+	// because the logger is not initialized yet
+	if err != nil && err != config.ErrMissingAPIKey {
```

Comment:
I was thinking more like
```suggestion
	if err != nil {
	    if err == config.ErrMissingAPIKey {
	        exitProcess = true
	   }
```
We still need the exit for the other cases.

Reply:
You're right. Sorry about that. This is getting a bit yucky with all the bools and if statements. Can't we just drop the logger and use `fmt.Println` just like `osutil.Exitf` does? I mean you only care about changing the exit code from 1 to 0.

We could factor out the graceful exiting into `osutil.ExitGraceful` or something